### PR TITLE
Transformer Interpretability Beyond Attention Visualization

### PR DESCRIPTION
**Pull request by nathanpainchaud**

### CODE REVIEW

**Review by HangJung97 — CHANGES_REQUESTED** [review]
Great review! Just a few typos.

File: collections/_posts/2023-04-24-TransformerInterpretability.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+\begin{align}
+\mathbf{\hat{A}}^{(b)} & = \mathbf{I} + \mathbb{E}_h \mathbf{A}^{(b)} \tag{6} \\
+\text{rollout} & = \mathbf{\hat{A}}^{(1)} \cdot \mathbf{\hat{A}}^{(2)} \cdot \ldots \cdot \mathbf{\hat{A}}^{(B)} \tag{7}
+\end{align}
+$$
+
+
+# Data
+The authors test their method on a vision task (segmentation of ImageNet-segmentation by thresholding the relevance maps
+by the mean value) and an NLP task (extraction of rationales supporting the binary sentiment classification on the Movie
```
[the remark this answers is not in the document]
Comment (nathanpainchaud) [reply collections/_posts/2023-04-24-TransformerInterpretability.md:109]: That was voluntary, and I would not consider this a typo :wink: Normally you use _an_ when the next word starts with a vowel sound (in this case _N_ is pronounced _[en]_) even if the letter itself is a consonant.

Here is a [post on the subject](https://www.scribbr.com/commonly-confused-words/a-vs-an/#:~:text=Revised%20on%20March%2011%2C%202023,%2C%E2%80%9D%20%E2%80%9Ci%E2%80%9D).) with the closest example I could find to the current case.

[the remark this answers is not in the document]
Comment (HangJung97) [reply collections/_posts/2023-04-24-TransformerInterpretability.md:109]: Thanks for pointing this out. Didn't know this applies to acronym as well.

File: collections/_posts/2023-04-24-TransformerInterpretability.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+$$
+\begin{align}
+R^{(n)}_j & = \mathcal{G}_q(x,w,q,R^{(n-1)}) \\
+& = \sum_{ \{ i | (i,j) \in q \} } \frac{x_j w_{ji}}{ \sum_{ \{ j' | (j',i) \in q \} } x_{j'} w_{j'i} } R^{(n-1)}_i
+\end{align} \tag{3}
+$$
+
+To initialize the relevance propagation, the relevance of $$R^{(0)}$$, i.e. the output layer, is set to a one-hot
+encoding of the class for which of the class w.r.t. which the relevance should be computed.
```
[the remark this answers is not in the document]
Comment (nathanpainchaud) [reply collections/_posts/2023-04-24-TransformerInterpretability.md]: That depends what you mean by _target_. It's not target in the sense of the groundtruth class for the current input, but rather in the sense of the target class for which you want to measure the network's attention.

But other than that small clarification, your interpretation is spot on! I also noticed that the end of the sentence is all mangled up (a mash up of different ways I tried to formulate this), so it was not easy decipher. I've rephrased things to be clearer :wink:

[the remark this answers is not in the document]
Comment (HangJung97) [reply collections/_posts/2023-04-24-TransformerInterpretability.md]: For _target_, I mean the class you want to measure the attention. The rephrased sentence looks good and clear to me.

**Review by HangJung97 — APPROVED** [review]

**Review by OlivierBernard — APPROVED** [review]
Excellent review ! I found one typo, but on a figure ;)))

**Pull request merged**